File: features/lookin/ts/Lookin.ts
import { int64, int32 } from "./lib/less";

export enum GroupBy {
    none = "none",
    uid = "uid",
    fuid = "fuid"
}

/**
 * 在看
 * @type db
 */
export class Lookin {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 目标ID
     * @index ASC
     */
    tid: int64 = 0

    /**
     * 项ID
     * @index ASC
     */
    iid: int64 = 0

    /**
     * 用户ID
     * @index ASC
     */
    uid: int64 = 0

    /**
     * 好友ID
     * @index ASC
     */
    fuid: int64 = 0

    /**
     * 好友推荐码
     */
    fcode: string = ""

    /**
     * 关系级别
     * @index ASC
     */
    flevel: int32 = 0

    /**
     * 其他数据
     * @length -1
     */
    options: any

    /**
     * 创建时间
     */
    ctime: int64 = 0

}

File: features/lookin/lookin/Lookin.go
package lookin

import (
	"github.com/hailongz/golang/db"
)

type Lookin struct {
	db.Object
	Tid	int64	`json:"tid" name:"tid" title:"目标ID" index:"ASC"`
	Iid	int64	`json:"iid" name:"iid" title:"项ID" index:"ASC"`
	Uid	int64	`json:"uid" name:"uid" title:"用户ID" index:"ASC"`
	Fuid	int64	`json:"fuid" name:"fuid" title:"好友ID" index:"ASC"`
	Fcode	string	`json:"fcode" name:"fcode" title:"好友推荐码"`
	Flevel	int32	`json:"flevel" name:"flevel" title:"关系级别" index:"ASC"`
	Options	interface{}	`json:"options,omitempty" name:"options" title:"其他数据" length:"-1" jsonType:"true"`
	Ctime	int64	`json:"ctime" name:"ctime" title:"创建时间"`
}

func (O *Lookin) GetName() string {
	return "lookin"
}

func (O *Lookin) GetTitle() string {
	return "在看"
}

